### ISSUE TITLE
Feature request: expm1

### ISSUE THREAD

**Issue by frkl** [pytorch/pytorch#4007]
Hi,

numpy has expm1 and log1p functions for numerically stable exp(x)-1 and log(1+x) when x is small. For instance, expm1 can be computed using

`exp(x)-1=x+x/2!+x/3!+x/4!+...`

There should be faster implementations out there somewhere.

These two functions are useful for numerically accurate conversions between log probability and logit. Below are my current mock up implementations. They are very slow and might use huge speedup with a proper implementation.

```
def expm1(x): #only works in ~ [-0.5,0.5]
	v=x;
	v=(v/8+1)*x
	v=(v/7+1)*x
	v=(v/6+1)*x
	v=(v/5+1)*x
	v=(v/4+1)*x
	v=(v/3+1)*x
	v=(v/2+1)*x
	return v;

def log1p(x): #only works in ~ [-0.5,0.5]
	v=x/10;
	v=-v*x+x/9;
	v=-v*x+x/8;
	v=-v*x+x/7;
	v=-v*x+x/6;
	v=-v*x+x/5;
	v=-v*x+x/4;
	v=-v*x+x/3;
	v=-v*x+x/2;
	v=-v*x+x;
	return v;
```

Also, it would be nice to add functions that convert log probability to logit and back. 
Converting logit to log probability is logsigmoid. Current implementation of logsigmoid has good precision but does not support second order gradient. Converting log probability to logit is logitexp. 
Here are my mock up implementations.

```

def logsigmoid(x):
	pos=torch.clamp(x,min=1);
	neg=torch.clamp(x,max=1);
	out=-log1p(torch.exp(-pos))+neg-torch.log(1+torch.exp(neg))+0.31326168751;
	return out;

def logitexp(logp):
	pos=torch.clamp(logp,min=-0.69314718056);
	neg=torch.clamp(logp,max=-0.69314718056);
	neg_val=neg-torch.log(1-torch.exp(neg));
	pos_val=-torch.log(torch.clamp(expm1(-pos),min=1e-20));
	return pos_val+neg_val;
```

**Comment by SsnL** [comment]
Commenting on `log1p`, it is available here http://pytorch.org/docs/master/torch.html#torch.log1p. :)

**Comment by vishwakftw** [comment]
`torch.expm1` has been merged into master. I think this issue can be closed now.
